### PR TITLE
Introduce Docker

### PR DESCRIPTION
**Pull request by marusak**
This is not for production use, mostly for development.

### CODE REVIEW

File: docker/Dockerfile_local [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+# From not on work from retrace-server directory
+WORKDIR '/retrace-server'
+
+# Change owner of /retrace-server, clean git and install dependences
+RUN chown -R --silent retrace /retrace-server && \
+    chmod -R --silent g=u /retrace-server && \
+    dnf -y install rpm-build sudo git &&  \
+    git clean -dfx && \
+    eval dnf -y --setopt=strict=0 --setopt=tsflags=nodocs install $(./autogen.sh sysdeps)
```
Comment (xsuchy) [inline docker/Dockerfile_local]: add "dnf clean all" here?

File: docker/Dockerfile_local [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+RUN rpm -Uvh noarch/retrace-server-* && \
+    /usr/libexec/fix-permissions /retrace-server && \
+    /usr/libexec/fix-permissions /var/log/retrace-server && \
+    /usr/libexec/fix-permissions /var/spool/retrace-server && \
+    sed -i -e"s/AllowTaskManager\s*=\s*0/AllowTaskManager = 1/i" /etc/retrace-server.conf && \
+    sed -i -e"s/RequireGPGCheck\s*=\s*1/RequireGPGCheck = 0/i" /etc/retrace-server.conf && \
+    mkdir /var/tmp/local_repo && \
+    dnf --releasever=25 --enablerepo=\*debuginfo\* -y --installroot=/var/tmp/local_repo/ \
+    download --resolve --destdir /var/tmp/local_repo/ abrt-addon-ccpp shadow-utils \
+    gdb rpm will-crash
```
Comment (xsuchy) [inline docker/Dockerfile_local]: add "dnf clean all" here?

File: docker/Dockerfile_local [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+
+# Update FAF
+RUN rpm -Uvh noarch/retrace-server-* && \
+    /usr/libexec/fix-permissions /retrace-server && \
+    /usr/libexec/fix-permissions /var/log/retrace-server && \
+    /usr/libexec/fix-permissions /var/spool/retrace-server && \
+    sed -i -e"s/AllowTaskManager\s*=\s*0/AllowTaskManager = 1/i" /etc/retrace-server.conf && \
+    sed -i -e"s/RequireGPGCheck\s*=\s*1/RequireGPGCheck = 0/i" /etc/retrace-server.conf && \
+    mkdir /var/tmp/local_repo && \
+    dnf --releasever=25 --enablerepo=\*debuginfo\* -y --installroot=/var/tmp/local_repo/ \
```
Comment (xsuchy) [inline docker/Dockerfile_local]: 25 hardcoded?

File: docker/Dockerfile_local (hunk 1 above)
Comment (marusak) [reply docker/Dockerfile_local]: No need, as dnf is run twice and local builds are only for development purposes where speed of build is more important than the size of image

**Comment by marusak** [comment]
@xsuchy updated

**Pull request merged**